src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTest.java

```diff
@@ -523,4 +524,214 @@ public void testApi_RemoveInspectionPort_InspectionPortDisappears() throws Excep
 
         assertEquals(null, foundInspectionPort);
     }
+
+    @Test
+    public void testApiRegisterPort() throws Exception {
+        //Arrange
+        port.setParentId(UUID.randomUUID().toString());
+        port.setDeviceOwnerId("DeviceOwnerId");
+
+        //Act
+        NetworkElement registeredPort = this.redirApi.registerPort(port);
+
+        //Assert
+        assertNotNull(registeredPort);
+        assertNotNull(registeredPort.getElementId());
+        assertNotNull(registeredPort.getParentId());
+        assertNotNull(registeredPort.getMacAddresses());
+        assertNotNull(registeredPort.getPortIPs());
+    }
+
+    @Test
+    public void testApiDeletePort() throws Exception {
+        //Arrange
+        port.setParentId(UUID.randomUUID().toString());
+        port.setDeviceOwnerId("DeviceOwnerId");
+
+        this.redirApi.registerPort(port);
+
+        //Act
+        this.redirApi.deletePort(port.getElementId());
+
+        //Assert
+        assertEquals(null, this.redirApi.getPort(port.getElementId()));
+    }
+
+    @Test
+    public void testGetPortIds() throws Exception {
+        //Arrange
+        this.redirApi.registerPort(ingress);
+        this.redirApi.registerPort(egress);
+        this.redirApi.registerPort(port);
+
+        //Act
+        List<String> portIds = this.redirApi.getPortIds();
+
+        //Assert
+        assertNotNull(portIds);
+        assertEquals(3, portIds.size());
+        assertTrue(portIds.contains(ingress.getElementId()));
+        assertTrue(portIds.contains(egress.getElementId()));
+        assertTrue(portIds.contains(port.getElementId()));
+    }
+
+    @Test
+    public void testApiGetPort() throws Exception {
+        //Arrange
+        port.setParentId(UUID.randomUUID().toString());
+        port.setDeviceOwnerId("DeviceOwnerId");
+
+        assertNotNull(this.redirApi.registerPort(port));
+
+        //Act
+        PortEntity registeredPort = this.redirApi.getPort(port.getElementId());
+
+        //Assert
+        assertNotNull(registeredPort);
+        assertNotNull(registeredPort.getElementId());
+        assertNotNull(registeredPort.getParentId());
+        assertNotNull(registeredPort.getMacAddresses());
+        assertNotNull(registeredPort.getPortIPs());
+    }
+
+    @Test
+    public void testApiUpdatePort() throws Exception {
+        //Arrange
+        final ArrayList<String> macAdresses = new ArrayList<String>(
```

Comment:
you can declare all final variable as class instances

please address multiple places

Reply:
removed final and retained as local variables in the method.